### PR TITLE
more e2eservice.TestJig cleanups

### PR DESCRIPTION
**Pull request by danwinship**
**What this PR does / why we need it:**
A bit more `e2eservice` cleanup: When `test/e2e/framework/service` was first created, a lot of service test code got crammed into `TestJig` even though it didn't use any of `TestJig`'s members. Make that code available outside of `TestJig`.

(There are other functions in `TestJig` that don't currently make much use of it, but I have further cleanup plans that will make them fit better there.)

/kind cleanup
/priority important-longterm
/sig network

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
+1 on the cleanup. CI seems to be reporting some errors (e.g. the bazel jobs).

**Comment by danwinship** [comment]
@neolit123 build error fixed, tests pass now

**Comment by neolit123** [comment]
/assign @andrewsykim 
PTAL for /test approval. thanks!

File: test/e2e/framework/service/affinity_checker.go [hunk 1]
```diff
@@ -17,9 +17,68 @@ limitations under the License.
...
+	"k8s.io/kubernetes/test/e2e/framework"
 	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
 )
 
+// CheckAffinity function tests whether the service affinity works as expected.
+// If affinity is expected, the test will return true once affinityConfirmCount
+// number of same response observed in a row. If affinity is not expected, the
+// test will keep observe until different responses observed. The function will
+// return false only in case of unexpected errors.
+func CheckAffinity(execPod *v1.Pod, targetIP string, targetPort int, shouldHold bool) bool {
```
Comment (timothysc) [inline test/e2e/framework/service/affinity_checker.go]: So I don't think this shuffling makes sense to this package.  I know they were lumped before, but as we are breaking apart the framework they should go into logical sub-packages that make sense to vendor independently.  I'd argue that this should really be in the pod library.

File: test/e2e/framework/service/resource.go [hunk 2]
```diff
@@ -116,3 +118,53 @@ func GetServiceLoadBalancerCreationTimeout(cs clientset.Interface) time.Duration
 	}
 	return LoadBalancerCreateTimeoutDefault
 }
+
+// CreatePausePodDeployment creates a deployment for agnhost-pause pod running in different nodes
+func CreatePausePodDeployment(cs clientset.Interface, name, ns string, replica int32) *appsv1.Deployment {
```
Comment (timothysc) [inline test/e2e/framework/service/resource.go]: this should be in the deployment library.

File: test/e2e/framework/service/util.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+		return false, nil
+	}); pollErr != nil {
+		e2elog.Failf("Could not reach HTTP service through %v:%v%v after %v: %v", host, port, url, timeout, pollErr)
+	}
+	return body
+}
+
+// GetNodes returns the first maxNodesForTest nodes. Useful in large clusters
+// where we don't eg: want to create an endpoint per node.
+func GetNodes(cs clientset.Interface, maxNodesForTest int) (nodes *v1.NodeList) {
```
Comment (timothysc) [inline test/e2e/framework/service/util.go]: This should not be in service.

**Review by timothysc — CHANGES_REQUESTED** [review]
/hold 

This is just a code shuffle vs. breaking up into logical units.  The work we are doing in the testing commons is to break apart the framework into logic sub-packages that can be be vendored whose transitive dep-graph is logical and sound.

**Comment by danwinship** [comment]
> This is just a code shuffle vs. breaking up into logical units.

Sure, this was not intended to be part of the e2e-wide cleanup effort (which I didn't even know was happening when I started this). I was just trying to make `e2eservice.TestJig` less of a dumpster fire while minimizing unrelated changes.

>  I'd argue that [`CheckAffinity`] should really be in the pod library.

It is testing Service-specific functionality (`SessionAffinity`). Maybe the `targetIP` and `targetPort` parameters should be renamed `serviceIP` and `servicePort` to make that clearer. (It can't just take a `v1.Service` as the argument because on some tests we want to use the service's cluster IP but in others we want to use an ingress IP.)

> [`CreatePausePodDeployment`] should be in the deployment library.

Hm... it looks like `e2edeployment.NewDeployment` covers 80% of it, and the rest is very specific to the 2 `test/e2e/network/service.go` tests that currently use it, so that could be put in a small helper function there.

> [`GetNodes` with `maxNodes`] should not be in service.

Yes, it looks like the idea of `GetReadySchedulableNodesOrDie`-but-not-too-many pops up in a bunch of places throughout e2e...

**Comment by andrewsykim** [comment]
Agreed with Tim that the Deployment / Node utilities should probably be somewhere else (though to be fair they already exist in the TestJig so maybe it's okay to address this in a follow-up PR), but +1 on decoupling generally usable functions from Service TestJig.

**Comment by danwinship** [comment]
- rebased around `e2elog` removal
- renamed `CheckAffinity` args to make it clearer that it's service-specific
- made `CreatePausePodDeployment` a local function in `test/e2e/network/service.go` (and used `e2edeployment` helper functions)
- dropped the changes related to `GetNodes`/`GetNodesNames`, which will instead be removed by #82291

**Pull request merged**